apps/ensadmin-next/src/components/connections/connection-selector.tsx

```diff
@@ -29,7 +30,10 @@ export function ConnectionSelector() {
   const handleSelect = (url: string) => {
     const params = new URLSearchParams(searchParams);
     params.set("ensnode", url);
-    router.push(`?${params.toString()}`);
+
+    router.push(`${pathname}?${params.toString()}`);
+
+    window.dispatchEvent(new CustomEvent("ensnode/connection/set", { detail: { url } }));
```

Comment:
This event allows the Next.js error boundary to be notified so it can attempt a recovery.